### ISSUE TITLE
MOC computation reads same climatology for separate test cases

### ISSUE THREAD

**Issue by vanroekel** [MPAS-Dev/MPAS-Analysis#144]
On anvil I was comparing the MOC from two different G-cases.  After the first test case computed and wrote the files similar to mocTimeSeries_year0002.nc, I changed the name and run directory to the second case and the second case read the same mocTimeSeries files.  I don't think this is the correct behavior and these files should have some sort of unique label or folder corresponding to the run name.

**Comment by milenaveneziani** [comment]
@vanroekel: did you also change the basedirectory for your analysis output?
The output files are stored in analysis_directory/timeseries, and, similarly to other filenames, they don't have the runname information in their names.

**Comment by vanroekel** [comment]
No I did not.  I wasn't aware this was required.  It was not previously.

**Comment by milenaveneziani** [comment]
I guess we need to decide how we want to handle this, now that we are saving analysis information to file. We talked about this with @xylar at one point.
One option could be to include the runname in the output files, of course.

**Comment by vanroekel** [comment]
What about appending mainRunName to baseDirectory in run_analysis.py such that baseDirectory = baseDirectory + '/' + mainRunName?  Seems like a relatively quick fix to this issue.

**Comment by xylar** [comment]
@vanroekel, I definitely think the expected usage is to run with a new output directory each time you run the analysis unless you are re-analyzing the same run.

**Comment by xylar** [comment]
Clearly, we need to document this better in both the readme and config.default. It would be great if someone could make a PR with a suggested change​.

**Comment by xylar** [comment]
@vanroekel, I really would prefer not to have something appended because I organize my output first by run, then by what branch I'm testing.

I haven't played with it but it should be possible for *you* to include a reference to the config option mainRunName in your definition of the base directory name within your customized config file. I will look into how to do that.

**Comment by vanroekel** [comment]
Okay, fair enough.  Thanks for looking into the change.

On Mon, Mar 20, 2017 at 11:02 AM, Xylar Asay-Davis <notifications@github.com

**Comment by vanroekel** [comment]
@xylar it looks like the most straightforward implementation of what you mention above is introduced in python 3.2.  For python 2, I had to do this

```
[DEFAULT]

runName = 20170221.beta0.GMPAS-IAF.T62_oECv3.anvil
[runs]
## options related to the run to be analyzed and reference runs to be
## compared against

# mainRunName is a name that identifies the simulation being analyzed.
mainRunName = %(runName)s

# referenceRunName is the name of a reference run to compare against (or None
# to turn off comparison with a reference, e.g. if no reference case is
# available)
referenceRunName = None
# preprocessedReferenceRunName is the name of a reference run that has been
# preprocessed to compare against (or None to turn off comparison).  Reference
# runs of this type would have preprocessed results because they were not
# performed with MPAS components (so they cannot be easily ingested by
# MPAS-Analysis)
preprocessedReferenceRunName = None

[input]
## options related to reading in the results to be analyzed

# directory containing model results
baseDirectory = /lcrc/group/acme/jwolfe/acme_scratch/20170221.beta0.GMPAS-IAF.T62_oECv3.anvil/run
# names of namelist and streams files.  If not in baseDirectory, give full path
oceanNamelistFileName = mpas-o_in
oceanStreamsFileName = streams.ocean
seaIceNamelistFileName = mpas-cice_in
seaIceStreamsFileName = streams.cice

# names of ocean and sea ice meshes (e.g. EC60to30, QU240, RRS30to10, etc.)
mpasMeshName = mesh

[output]
## options related to writing out plots, intermediate cached data sets, logs,
## etc.

# directory where analysis should be written
# Must be changed for each run
# One possible method
# /path/to/base/directory/%(runName)s
baseDirectory = /lcrc/group/acme/lvanroe/%(runName)s
```

I'm not sure that this is a good change as I'm not sure if it functions with ACME interfaces this way.  If you find something better, let me know.  For now I will use this as a personal config file.

**Comment by xylar** [comment]
So to follow up, there is a syntax for referencing config options in the same section:
```
baseDirectory = blah
runDirectory: %(baseDirectory)/run
```
In python 3.2 and above, there is a much more sophisticated syntax that would allow us to reference options in other sections.  Obviously, for now we are using 2.7.

**Comment by vanroekel** [comment]
You beat me to the comment by a few seconds.  I found the same issue.

**Comment by xylar** [comment]
Yeah, I don't think using DEFAULT is a good solution for us.

**Comment by xylar** [comment]
@vanroekel, I think your best bet might be a mini script that modifies the config file using mainRunName wherever you want it before running...

**Comment by vanroekel** [comment]
@xylar good idea.  I will do that.  As for making the need to change baseDirectory in [output], should we just modify [output] to say something like

[output]
## options related to writing out plots, intermediate cached data sets, logs,
## etc.

# directory where analysis should be written
# NOTE: This directory path must be specific to each test case.

**Comment by milenaveneziani** [comment]
Sounds good, @vanroekel. How about also modifying point 2. in README?
thanks.
